### PR TITLE
test: fix a node 10 test failure

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This fixes the failing nightly test.

### CODE REVIEW

File: test/unit/google-apis/auth-client.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ function verifyReportedMessage(
     constructor() {
       this.authClient = {
         async getAccessToken() {
-          throw errToReturn;
+          return Promise.reject(errToReturn);
```
Comment (ofrobots) [inline test/unit/google-apis/auth-client.ts]: I don't quite understand what is changing. The two sides look mostly equivalent to me. How does this fix the failing test?

Comment (JustinBeckwith) [reply test/unit/google-apis/auth-client.ts]: I was throwing things against the wall to see why the test was failing.  This seemed to work, but now I'm thinking it's a flaky test. `npm test` totally Works on My Machine ™️.  I will keep digging here.

Comment (ofrobots) [reply test/unit/google-apis/auth-client.ts]: To elaborate, throwing inside an async function is equivalent to returning a rejected promise. I might be missing something though.

Comment (JustinBeckwith) [reply test/unit/google-apis/auth-client.ts]: Ok, I have a theory.  I think this test was on the edge of timing out.  If it throws an exception due to timeout, it was getting captured somehow, and causing a crash with a `Cannot read property 'message' of null` message:
https://circleci.com/gh/googleapis/nodejs-error-reporting/4790

I suspect the change to `Promise.reject` worked randomly because the test was on the edge of timing out, and happened to land within 4 seconds that time 😆 

After the bumping the timeout up to 8s, and rollin back my `Promise.reject` change, it seems to be running stable.  

TL;DR - test was timing out, Justin dum
